### ISSUE TITLE
[MU4 Issue] Hiding and showing the tuplet button causes it to appear disabled

### ISSUE THREAD

**Issue by bkunda** [musescore/MuseScore#10888]
**Describe the bug**
When the tuplet button in the note input toolbar is hidden then shown again, it returns in an apparently disabled (greyed-out) state. 

![disabled-tuplet-icon](https://user-images.githubusercontent.com/86290556/159927150-ab1b8cee-878c-4ecd-b7b4-6c5426e2ebc4.gif)

This doesn't affect its functionality, but it does present as a bit strange because the behaviour is not replicated across other buttons in the toolbar. 

**Expected behavior**
Triplet icon shouldn't appear greyed out when changing visibility states.
When nothing is selected in the score, clicking on the tuplet button should perform no action (as per current behaviour of "Tie", "Slur" and "Flip direction" buttons).

**Platform information**
 - MacOS 12.1 (M1)
- Windows 11 

**Note**
The weird scrolling behaviour you see will be the subject of another issue (@Tantacrul FYI)

**Comment by HemantAntony** [comment]
@bkunda Isn't this the desired behaviour? The tuplet button will become enabled when you click a measure, a rest, etc.. It makes sense too. You clicked on the page last. So, where would MuseScore put the tuplet

**Comment by bkunda** [comment]
I guess what troubles me about this one is that nothing else in the note input toolbar behaves like this. Take for example ties, slurs, flip object, etc. If the logic is that a toolbar button is only enabled when a score object is selected, then really this should apply across the entire toolbar. But then the entire toolbar would appear greyed out unless you selected something in the score (which we obviously don't want). 

In saying this, it's not a big problem the way it is – because the button becomes active when you actually need to use it (i.e. when something is selected). The issue really is a UI one – I just don't see why this button should behave differently to the rest. Hence I've marked it as a pretty low priority 🙂

**Comment by Eism** [comment]
see https://github.com/musescore/MuseScore/issues/9847
maybe it will help to understand why we added this behavior

**Comment by bkunda** [comment]
That does help. Thanks @Eism. 
I'll close this issue for now and discuss with @Tantacrul later.

**Comment by Tantacrul** [comment]
We added this behaviour because the tuplet button kept throwing an error whenever a user selected it (when nothing else was selected).

Perhaps what we can do instead is simply **not throw an error**. If the user has nothing selected on the score, then clicking the tuplet button simply does nothing.

Agreed @bkunda? (if so, reopen, I think :) )

**Comment by bkunda** [comment]
Yes I agree this makes sense. It's perfectly in line with the behaviour from the tie, slur, and flip direction buttons – none of which engage note input mode. I would put tuplets in the same category.

Reopening this issue and adding detail to "expected behaviour" in the issue description.

**Comment by Eism** [comment]
We have a "Show this message again" checkbox in such dialogs. If the user is tired of seeing such a message(at best, he understood how it works), then he will choose not to show this dialog anymore.
See https://github.com/musescore/MuseScore/issues/9668 for how we fixed this issue for the add menu. Maybe do the same for the note input toolbar?

PS: Hm, what the user should do to see this dialog again? For now, the user can see this dialog only after reverting to factory settings.

**Comment by bkunda** [comment]
@Eism we don't currently have this alert/dialog for other buttons in the toolbar (as far as I'm aware). So I'm not sure we really need it for this button either.

**Comment by Tantacrul** [comment]
Yeah, I think the message is unnecessary and a bit annoying. If the button doesn't do anything, then users will understand that right away. I'd just not show it at all.
